### ISSUE TITLE
Services not starting for user

### ISSUE THREAD

**Issue by PrasadChinwal** [Homebrew/homebrew-services#433]
| package | version |
| ---------| --------|
| macos | Monterey 12.1 |
| Homebrew| 3.3.9|
| php | 8.1.1 |
| nginx | 1.21.5 |
| dnsmasq | 2.86 |

### Services not started when `brew services start <service_name>` is run.

I recently started getting issues when running my services using brew services. Weird thing is when i run for ex. `brew services start php` I get the following error:
```
Bootstrap failed: 5: Input/output error
Try re-running the command as root for richer errors.
Error: Failure while executing; `/bin/launchctl bootstrap gui/501 /Users/prasadchinwal/Library/LaunchAgents/homebrew.mxcl.php.plist` exited with 5.
```

When I run this as sudo the message changes to:
<details>
  <summary>Message:</summary>
  
  ```
Warning: Taking root:admin ownership of some php paths:
  /usr/local/Cellar/php/8.1.1/sbin
  /usr/local/Cellar/php/8.1.1/sbin/php-fpm
  /usr/local/opt/php
  /usr/local/opt/php/bin
  /usr/local/opt/php/sbin
  /usr/local/var/homebrew/linked/php
This will require manual removal of these paths using `sudo rm` on
brew upgrade/reinstall/uninstall.
Warning: php must be run as non-root to start at user login!
==> Successfully started `php` (label: homebrew.mxcl.php)
```
</details>

# My findings:

<details>
  <summary>Results of `brew services list`</summary>
  
  <img width="866" alt="Screen Shot 2022-01-02 at 6 17 13 PM" src="https://user-images.githubusercontent.com/7385673/147893038-8c8b95c7-8003-4c34-b57a-03f5d2393ac5.png">
</details>

<details>
  <summary>Results of `sudo brew services list`</summary>
   
   <img width="740" alt="Screen Shot 2022-01-02 at 6 18 35 PM" src="https://user-images.githubusercontent.com/7385673/147893069-6b47f56a-4eb7-4a97-b21e-87cf19b438f5.png">
</details>

<details>
  <summary>Result for `launchctl list` </summary>
   
   <img width="404" alt="Screen Shot 2022-01-02 at 6 19 10 PM" src="https://user-images.githubusercontent.com/7385673/147893078-88dac39b-2ee0-4bf7-8b83-e05b0449d6f6.png">
</details>

<details>
  <summary>Result for `sudo launchctl list` </summary>
   
  <img width="401" alt="Screen Shot 2022-01-02 at 6 20 01 PM" src="https://user-images.githubusercontent.com/7385673/147893091-237cb78c-7622-4228-ad9e-c313a4c09b32.png">
</details>

It shows that nginx is started and has a pid but other services like php, dnsmasq don't have a pid.

I looked up a lot and found that this might be due to 2 separate .plist files registered. I tried to delete the .plist files from `/Library/LaunchAgents/` location. But this was no help.

Tried to run `brew services cleanup` but that didn't do the trick as well.

Kindly let me know if more information is required. Any help is appreciated.

**Comment by SMillerDev** [comment]
Can you try running `brew services stop php` as both user and root and then trying to run it again from only one system.

**Comment by PrasadChinwal** [comment]
Hey @SMillerDev, I followed your suggestion and it stopped both services. Unfortunately, it didn't start the services.
<details>
  <summary>Results</summary>
  
```
    ❯ brew services stop php
Stopping `php`... (might take a while)
==> Successfully stopped `php` (label: homebrew.mxcl.php)
❯ sudo brew services stop php
Password:
Sorry, try again.
Password:
Stopping `php`... (might take a while)
==> Successfully stopped `php` (label: homebrew.mxcl.php)
❯ brew services list
Name    Status  User          File
dnsmasq error  512 root          ~/Library/LaunchAgents/homebrew.mxcl.dnsmasq.plist
httpd   error  6               /usr/local/opt/httpd/homebrew.mxcl.httpd.plist
mailhog started               /usr/local/opt/mailhog/homebrew.mxcl.mailhog.plist
mysql   none
nginx   error  256 prasadchinwal ~/Library/LaunchAgents/homebrew.mxcl.nginx.plist
php     none
redis   stopped               /usr/local/opt/redis/homebrew.mxcl.redis.plist
unbound none
❯ sudo brew services list
Name    Status  User File
dnsmasq none root
httpd   none
mailhog none
mysql   none
nginx   started root /Library/LaunchDaemons/homebrew.mxcl.nginx.plist
php     none
redis   none
unbound none
❯ brew services start php
==> Successfully started `php` (label: homebrew.mxcl.php)
❯ brew services list
Name    Status  User          File
dnsmasq error  512 root          ~/Library/LaunchAgents/homebrew.mxcl.dnsmasq.plist
httpd   error  6               /usr/local/opt/httpd/homebrew.mxcl.httpd.plist
mailhog started               /usr/local/opt/mailhog/homebrew.mxcl.mailhog.plist
mysql   none
nginx   error  256 prasadchinwal ~/Library/LaunchAgents/homebrew.mxcl.nginx.plist
php     error  19968 prasadchinwal ~/Library/LaunchAgents/homebrew.mxcl.php.plist
redis   stopped               /usr/local/opt/redis/homebrew.mxcl.redis.plist
unbound none
  ```
</details>

**Comment by SMillerDev** [comment]
Check `brew services info php` and see if there are errors in the log

**Comment by PrasadChinwal** [comment]
I didn't find any error logs but here are the results. I checked the `launchctl` to see if the pid was assigned since `brew services start php` says successfully started but the pid was not assigned.

<details>
 <summary>Results</summary>

 ```
❯ brew services info php
php (homebrew.mxcl.php)
Running: ✘
Loaded: ✘
Schedulable: ✘
❯ brew services start php
==> Successfully started `php` (label: homebrew.mxcl.php)
❯ brew services info php
php (homebrew.mxcl.php)
Running: ✘
Loaded: ✔
Schedulable: ✘
❯ launchctl list | grep homebrew.
-	2	homebrew.mxcl.dnsmasq
-	1	homebrew.mxcl.nginx
-	78	homebrew.mxcl.php
-	0	homebrew.mxcl.redis
-	78	homebrew.mxcl.php@8.0
-	-6	homebrew.mxcl.httpd
50346	-15	homebrew.mxcl.mailhog
❯ sudo launchctl list | grep homebrew.
30064	0	homebrew.mxcl.nginx
-	78	homebrew.mxcl.php@8.0
  ```
</details>

Is this related to this issue #102 ?

**Comment by PrasadChinwal** [comment]
@SMillerDev I figured out the issue. 

I had to follow the below steps to solve this issue.

- Uninstalled all services from brew one by one.
- Deleted all `.plist` from locations `~/Library/LaunchAgents` and `/Library/LaunchAgents` and `~/Library/LaunchDaemons`.
- Deleted all config files for the services from respective locations. ex. `nginx` config files (my location was `/usr/local/etc/nginx`) and/or `php`.
- Reinstalled required services.

What I believed happened is when I upgraded brew manually the previous config and .plist files for services were stopping the services from running. `brew services start php` showed status running but digging into `launchctl` helped me in debugging the issue.

Thanks for the help 👍
